.github/workflows/docs-site.yml

```diff
@@ -0,0 +1,94 @@
+# Generates and deploys static site to GitHub Pages
+# https://docs.github.com/en/pages/getting-started-with-github-pages/using-custom-workflows-with-github-pages
+# Reference workflow: https://github.com/actions/starter-workflows/blob/main/pages/jekyll-gh-pages.yml
+name: Metro Docs Site
+
+on:
+  push:
+    branches: ["main"]
+
+  # Allows manual deployements (if needed)
+  workflow_dispatch:
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
+# However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
+concurrency:
+  group: "pages"
+  cancel-in-progress: false
+
+jobs:
+  # Metro docs site build job
+  docs-site:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4 # Or, actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
```

Comment:
Not sure why specific hash was used for this action! Should I match it?

Reply:
renovate will use specific hashes by default for idempotence

Reply:
```suggestion
        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
```